src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -1960,61 +1991,67 @@ private List<CompletionResult> GetResultForIdentifier(CompletionContext completi
                 // Handle completion for a path with variable, such as: $PSHOME\ty<tab>
                 if (completionContext.RelatedAsts.Count > 0 && completionContext.RelatedAsts[0] is ScriptBlockAst)
                 {
-                    Ast cursorAst = null;
-                    var cursorPosition = (InternalScriptPosition)_cursorPosition;
-                    int offsetBeforeCmdName = cursorPosition.Offset - tokenAtCursorText.Length;
-                    if (offsetBeforeCmdName >= 0)
-                    {
-                        var cursorBeforeCmdName = cursorPosition.CloneWithNewOffset(offsetBeforeCmdName);
-                        var scriptBlockAst = (ScriptBlockAst)completionContext.RelatedAsts[0];
-                        cursorAst = GetLastAstAtCursor(scriptBlockAst, cursorBeforeCmdName);
-                    }
```

Comment:
Can you please explain the reason for the changes here? Also, would be nice if you add comment about what cases for indexing completion is covered here.

Reply:
The design was flawed (probably because it was written in a time before PSReadline and ISE). It assumed the cursor would be at the end of the word like `$PSVersionTable[psver<Tab>` so if you moved the cursor back towards the beginning of the word it would fail.  
The consolehost tab completion implementation without PSReadline trims everything after the cursor when doing completions so it was probably valid code when it was written.

I've added comments that explain all the different cases.